lib/galaxy/managers/collections.py

```diff
@@ -592,18 +599,23 @@ def _build_elements_from_rule_data(self, collection_type_description, rule_set,
                         found = True
 
                     if not found:
-                        sub_collection = {}
+                        sub_collection: Dict[str, Any] = {}
                         sub_collection["src"] = "new_collection"
                         sub_collection["collection_type"] = collection_type_at_depth.collection_type
                         sub_collection["elements"] = {}
-                        elements_at_depth[identifier] = sub_collection
-                        elements_at_depth = sub_collection["elements"]
+                        elements_at_depth[
+                            identifier
+                        ] = sub_collection  # ??? but this is overridenn on the next line??
+                        elements_at_depth = sub_collection[
+                            "elements"
+                        ]  # ??? sub_collection is just lost?
```

Comment:
Yes, this is a cry for help

Reply:
Yeah, I think all of this (L602-L611) could be replaced with `elements_at_depth = {}`, and we'll see if any tests break.